File: app/src/main/java/org/cuieney/videolife/presenter/contract/ToutiaoVideoHomeContract.java
package org.cuieney.videolife.presenter.contract;

import org.cuieney.videolife.common.base.BasePresenter;
import org.cuieney.videolife.common.base.BaseView;
import org.cuieney.videolife.entity.ToutiaoBean.MultiNewsArticleDataBean;

import java.util.List;

/**
 * Created by daimaren on 2017/10/13
 * 975808394@qq.com
 */

public interface ToutiaoVideoHomeContract {
	interface View extends BaseView{
		void showContent(List<MultiNewsArticleDataBean> dataBeen);
		void error(Throwable throwable);
	}
	interface Presenter extends BasePresenter<View>{
		/**
		 * 获取视频数据
		 */
		void getVideoData(String maxBehotTime, List<MultiNewsArticleDataBean> dataBeen);
	}
}

File: app/src/main/java/org/cuieney/videolife/kotlin/presenter/contract/ToutiaoVideoHomeContract.kt
package org.cuieney.videolife.kotlin.presenter.contract

import org.cuieney.videolife.entity.ToutiaoBean.ToutiaoVideoListBean
import org.cuieney.videolife.kotlin.base.BasePresenter
import org.cuieney.videolife.kotlin.base.BaseView

/**
 * Created by cuieney on 2017/6/2.
 */
interface ToutiaoVideoHomeContract {
    interface View : BaseView {
        fun showContent(data: ToutiaoVideoListBean)
        fun error(throwable: Throwable)
    }

    interface Presenter : BasePresenter {
        fun getVideoData(date: String, view: View)
    }
}